docs(test): moving all tests to dspo (#907)
opendatahub-io/data-science-pipelines-operator: proposals/bring_your_own_argo_workflows/byoaw_test_plan.md

| **Expected Results**  | <ul><li>Event conflicts properly identified</li><li>Clear ownership of workflow resources</li><li>No orphaned or stuck workflows</li><li>Predictable controller behavior documented</li></ul>                                                                                                                                                          |

### 3.2 Incompatible Argo Version

| Test Case ID          | TC-NF-002                                                                                                                                                                                            |
|-----------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify behavior with unsupported Argo versions                                                                                                                                                       |

> | **Test Steps**        | <ol><li> Install unsupported Argo version</li><li>Configure DSPA for external Argo</li><li>Attempt pipeline execution</li><li>Document error messages</li><li>Verify graceful degradation </li></ol> |

Reviewer: this pattern is everywhere in the testplan, so we should either update it in all instances, or create a sub-section (ie `Configure DSPO in BYOArgo Mode and Deploy a DSPA`) and refer to it

### 2.2 Complex Pipeline Types
Runs of different types of pipeline specs executes successfully. Pipelines that exercise all different inputs and outputs of a launcher/driver

| Test Case ID          | TC-PF-002                                                                                                                                                                                                                        |
|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify run of "Pipelines with artifacts" pipeline                                                                                                                                                                                |
| **Test Steps**        | <ol><li>Configure DSPA with external Argo </li><li> Execute pipeline - Pipelines with artifacts</li><li> Verify each pipeline type executes correctly</li><li> Validate artifacts, metadata, and custom configurations</li></ol> |

> | **Expected Results**  | <ul><li>Pipeline execute successfully</li><li>Artifacts are produced to the right s3 location and are consumed correctly </li></ul>                                                                                              |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Pipeline execute successfully</li><li>Artifacts are stored to the right s3 location and are consumed correctly </li></ul>                                                                                              |
```

| Test Case ID          | TC-PF-003                                                                                                                                                           |
|-----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify run of "Pipelines without artifacts" pipeline                                                                                                                |
| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Pipelines without artifacts</li><li> Verify each pipeline type executes correctly</li></ol> |                                                                                                                                                                                   |

> | **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>No artifacts are produced to S3</li></ul>                                                                                |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>No artifacts are stored into S3</li></ul>                                                                                |
```

| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Parallel DAGs running in parallel and completes successfully</li></ul>                                              |

| Test Case ID          | TC-PF-006                                                                                                                                                                                       |
|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify run of "Custom root KFP components" pipeline                                                                                                                                             |
| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Custom root KFP components </li><li> Verify each pipeline type executes correctly </li></ol>                            |

> | **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Artifcats are uploaded in the custom S3 bucket rather than the default, and downstream components are consuming from this custom location</li></ul>  |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Artifacts are uploaded to the custom S3 bucket rather than the default, and downstream components are consumed from this custom location</li></ul>  |
```

| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Pipelines with certificate handling </li><li>Verify each pipeline type executes correctly</li></ol> |
| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Components gets the right certificate installed </li></ul>                                                                       |

| Test Case ID          | TC-PF-017                                                                                                                                                               |
|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify run of "Conditional branching pipelines" pipeline                                                                                                                |
| **Test Steps**        | <ol><li> Configure DSPA with external Argo </li><li> Execute Pipeline - Conditional branching pipelines </li><li>Verify each pipeline type executes correctly</li></ol> |

> | **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Nested DAGs runs only if the expected condition is true</li></ul>                                                            |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Pipeline runs successfully</li><li>Nested DAGs run only if the expected condition is true</li></ul>                                                            |
```

| **Test Case Summary** | Verify DSP API Server compatibility with different external WorkflowController versions                                                                                                                                                                                                                                                                                                                                                                              |
| **Test Steps**        | <ol><li> Deploy DSP API Server with specific Argo library dependencies</li><li>Install external Argo WorkflowController with different version</li><li>Test API Server to WorkflowController communication</li><li>Verify Kubernetes API interactions (CRs, status updates)</li><li>Test pipeline submission, execution, and status reporting</li><li>Monitor for API compatibility issues or version mismatches</li><li>Document API compatibility matrix</li></ol> |

> | **Expected Results**  | <ul><li>API Server communicates successfully with external WC</li><li>Kubernetes API interactions work correctly</li><li>Pipeline lifecycle management functions properly</li><li>Status updates and monitoring work correctly</li><li>API compatibility documented and validated </li></ul>                                                                                                                                                                         |

Reviewer: I don't believe there's any direct communication between the APIServer and WorkflowController, per say, but just create and consume and update k8s objects like argoproj Workflows.  Might want to reword this test so that it doesn't seem there's a network interaction/connection between these two components

Author: Deleted this test case

|-----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Test Case Summary** | Verify DSPA uninstall behavior with external Argo                                                                                                                                                                                                                                                                                 |
| **Test Steps**        | <ol><li> Configure DSPA with external Argo (no internal WC)</li><li>Execute multiple pipelines and generate data</li><li>Delete DSPA</li><li>Verify external Argo WorkflowController remains intact</li><li>Verify DSPA-specific resources are cleaned up</li><li>Check that pipeline history is appropriately handled </li></ol> |

> | **Expected Results**  | <ul><li>DSPA removes cleanly</li><li>External Argo WorkflowController unaffected</li><li>No impact on other DSPAs using same external Argo</li><li>Pipeline data handling follows standard procedures </li></ul>                                                                                                                  |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>DSPA removed cleanly</li><li>External Argo WorkflowController unaffected</li><li>No impact on other DSPAs using same external Argo</li><li>Pipeline data handling follows standard procedures </li></ul>                                                                                                                  |
```

| **Expected Results**  | <ul><li>CRDs updated in-place successfully</li><li>Existing Workflows, WorkflowTemplates, CronWorkflows preserved</li><li>New CRs work with updated schema</li><li>No data loss or corruption</li><li>Compatibility maintained </li></ul>                                                                                                                                                                                   |

## 11 Initiative Level Tests
This is to verify that the integration of this feature with other product components does not introduce any regression. So this should be the very last tests that we need to run after verifying that there is no regression if used with last RHOAI release of other product components

| Test Case ID          | TC-IL-001                                                      |
|-----------------------|----------------------------------------------------------------|
| **Test Case Summary** | Verify that Iris Pipeline Runs on a **standard** RHOAI cluster |
| **Test Steps**        | <ol><li> Run an IRIS pipeline</li></ol>                        |

> | **Expected Results**  | Verify that the pipeline run succeeds </li></ul>               |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Verify that the pipeline run succeeds </li></ul>               |
```

| Test Case ID          | TC-IL-002                                                          |
|-----------------------|--------------------------------------------------------------------|
| **Test Case Summary** | Verify that Iris Pipeline Runs on a **FIPS Enabled** RHOAI cluster |
| **Test Steps**        | <ol><li> Run an IRIS pipeline</li></ol>                            |
| **Expected Results**  | Verify that the pipeline run succeeds </li></ul>                   |

| Test Case ID          | TC-IL-003                                                          |
|-----------------------|--------------------------------------------------------------------|
| **Test Case Summary** | Verify that Iris Pipeline Runs on a **Disconnected** RHOAI cluster |
| **Test Steps**        | <ol><li> Run an IRIS pipeline</li></ol>                            |

> | **Expected Results**  | Verify that the pipeline run succeeds </li></ul>                   |

Reviewer: ```suggestion
| **Expected Results**  | <ul><li>Verify that the pipeline run succeeds </li></ul>                   |
```
